### PR TITLE
chore(ci): add autofix

### LINKED ISSUE #2003: Use `autofix.ci` to fix code issues.
### What problem does this feature solve?

I observed the usage of the `autofix.ci` command in `slidev`, which enables automatic resolution of certain easily fixable issues, such as formatting problems. By utilizing the `just fix` command, open-source projects can freely access the `autofix.ci` command to promptly address these issues without the need to repeatedly employ the `just fmt` command in pull requests.

### What does the proposed API look like?

Only requires configuration of a CI pipeline.

### PR DESCRIPTION
**Pull request by shulaoda**
### Description

closes #2003

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -122,20 +122,6 @@ jobs:
       os: ${{ matrix.target }}
       changed: ${{ needs.changes.outputs.node-changes == 'true' }}
 
-  node-validation:
```
Comment (hyf0) [inline .github/workflows/ci.yml]: I think we still need this in case there are something wrong with autofix.

File: .github/workflows/autofix.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: autofix.ci # needed to securely identify the workflow
+
+on:
+  push:
+    branches:
+      - main
```
Comment (hyf0) [inline .github/workflows/autofix.yml]: Does autofix add extra commit? If so, it should not apply to main branch.

Comment (shulaoda) [reply .github/workflows/autofix.yml]: Let me experiment

File: .github/workflows/ci.yml (hunk 1 above)
Comment (shulaoda) [reply .github/workflows/ci.yml]: Do we still need to keep these?
https://github.com/rolldown/rolldown/blob/4c9feeb058f0212228e430450004967c99af9177/.github/workflows/ci.yml#L154-L158

Comment (IWANABETHATGUY) [reply .github/workflows/ci.yml]: Format checking could be removed since we have autofix.

**Review by IWANABETHATGUY — APPROVED** [review]
LGTM, cc @hyf0

File: .github/workflows/ci.yml (hunk 1 above)
Comment (hyf0) [reply .github/workflows/ci.yml]: Emm. I'm not sure about this. I consider autofix is just a robot that executing `just fix`  for you. But no matter who  are executing `just fix` we still need to validate the fix.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -151,12 +151,6 @@ jobs:
       - name: Lint Spell
         run: pnpm lint-spell
 
-      - name: Lint Prettier
```
Comment (hyf0) [inline .github/workflows/ci.yml]: Could you keep this? We still need these to validate if autofix is executing correctly.

Comment (shulaoda) [reply .github/workflows/ci.yml]: yes

**Review by hyf0 — APPROVED** [review]
This is cool.

**Pull request merged**